ouroboros-consensus-cardano/tools/db-validator/Main.hs

```diff
@@ -232,27 +280,31 @@ mkShelleyProtocolInfo genesis =
   CardanoBlock instance
 -------------------------------------------------------------------------------}
 
-instance TPraosCrypto c => HasProtocolInfo (CardanoBlock c) where
-  mkProtocolInfo [byronGenesis, shelleyGenesis] genesisHash requiresNetworkMagic = do
-    byronConfig <- openGenesisByron byronGenesis genesisHash requiresNetworkMagic
-    shelleyConfig <- either (error . show) return =<< Aeson.eitherDecodeFileStrict' shelleyGenesis
-    return $ mkCardanoProtocolInfo byronConfig shelleyConfig
-  mkProtocolInfo ls _ _ =
-    error $ "Two genesis files are needed for Cardano. Given " ++ show ls
+instance HasProtocolInfo (CardanoBlock TPraosStandardCrypto) where
+  type Args (CardanoBlock TPraosStandardCrypto) = CardanoBlockArgs
+  mkProtocolInfo CardanoBlockArgs {..} = do
+    let ByronBlockArgs {..} = byronArgs
+    let ShelleyBlockArgs {..} = shelleyArgs
+    byronConfig <- openGenesisByron configFileByron genesisHash requiresNetworkMagic
+    shelleyConfig <- either (error . show) return =<<
+      Aeson.eitherDecodeFileStrict' configFileShelley
+    return $ mkCardanoProtocolInfo byronConfig shelleyConfig threshold initialNonce
 
 mkCardanoProtocolInfo :: forall c. TPraosCrypto c
                       => CC.Genesis.Config
                       -> ShelleyGenesis c
+                      -> Maybe PBftSignatureThreshold
+                      -> Nonce
                       -> ProtocolInfo IO (CardanoBlock c)
-mkCardanoProtocolInfo byronConfig shelleyConfig =
+mkCardanoProtocolInfo byronConfig shelleyConfig signatureThreshold nonce =
     protocolInfoCardano
       byronConfig
-      (Just $ PBftSignatureThreshold 0.22)
+      signatureThreshold
       (CC.Update.ProtocolVersion 1 0 0)
```

Comment:
Should be `2 0 0` (see https://hydra.iohk.io/build/3670619/download/1/mainnet-config.json).